internal/static/metrics/wal/wal_test.go

```diff
@@ -787,6 +873,28 @@ func (b byRefSample) Less(i, j int) bool {
 	return b[i].Ref < b[j].Ref
 }
 
+type byRefHistogramSample []record.RefHistogramSample
+
+func (b byRefHistogramSample) Len() int      { return len(b) }
+func (b byRefHistogramSample) Swap(i, j int) { b[i], b[j] = b[j], b[i] }
+func (b byRefHistogramSample) Less(i, j int) bool {
+	if b[i].Ref == b[j].Ref {
+		return b[i].T < b[j].T
+	}
+	return b[i].Ref < b[j].Ref
+}
+
```

Comment:
Maybe we can reduce all the three sorting structs code by using https://pkg.go.dev/sort#Slice ?